src/cowboy_telemetry_h.erl

```diff
@@ -24,40 +18,83 @@ data(StreamID, IsFin, Data, [Next0 | StartTime]) ->
 
 info(StreamID, Info, [Next0 | StartTime]) ->
     {Commands, Next} = cowboy_stream:info(StreamID, Info, Next0),
```

Comment:
Is it possible for `info` to be called multiple times? Because if so, we need probably to add the `done` guard here too.

Reply:
It is called multiple times, but I'm unsure if it'd have `response` or `error_response` commands twice. I'll try to see if I can generate a test case...

Reply:
This led me to realize a different flow happens for chunked requests, so I've taken on a major refactor :) I really like the latest version, I followed the pattern in all the main cowboy handlers...

Reply:
I am a bit worried this change will force us to keep large chunks of the response in memory. I would prefer to not hold to any of it and instead emit the events as soon as possible.

Reply:
Good call, I can adjust things to avoid that. I was pushed in this direction to be able to handle a chunked response - there I do need to hold onto a small amount of state since the `headers` and the "end" come in different callbacks...

Reply:
I think we should then limit the amount of data we emit in callbacks. Chunked responses is one of the scenarios where you want to keep the amount of memory low, especially if you have thousands of conns streaming at the same time.

Reply:
The chunked response already will not emit any of the response chunks because of this very issue, we definitely don't want to accumulate those responses, that'd be a major problem

It does however, hold onto the `resp_status` and `resp_headers` to include them in the stop event...

Reply:
Ok, in this version, I use both a `state` and `acc` so that I can keep the data needed across handler callbacks in `state` and the data needed across the `fold` over `Commands` in `acc`... The `acc` is discarded after the `fold` so it shouldn't stick around longer than needed...

Reply:
Awesome job, looks good to me!

Reply:
Thanks!

I'll get this merged and released, then up next I'll start a conversation about some other measurements we might consider including...